docs/request.md

```diff
@@ -0,0 +1,156 @@
+
+`request()` provides additional request and response handling over `fetch`.
+
+For requests it:
+
+*   Prepends a configurable base URL to the endpoint you specify (see [Configuring a SDK object](./#configuring-a-sdk-object))
+*   Adds your [OAuth token](https://www.eventbrite.com/developer/v3/api_overview/authentication/#ebapi-getting-a-token) in the request `Authorization` header
+*   Sets the appropriate `Content-type` header depending on the `fetch` `method` (`GET`, `POST`, etc.) configuration you use
+*   Sets the appropriate `credentials` setting depending on the `fetch` `mode` (`cors`, etc.) configuration you use
+
+For responses it:
+
+*   Returns a resolved `Promise` with the response data parsed as JSON
+*   If the HTTP status is in the 400 or 500 range, returns a rejected `Promise` with parsed API errors, [if applicable](#error-handling)
+
+## API
+
+The TypeScript function definition of `request()` is:
+
+```
+(endpoint: string, options?: RequestInit): Promise<{}>
+```
+
+### Parameters
+
+`request()` accepts the following parameters:
+
+*   `endpoint`: The Eventbrite v3 API endpoint path, such as `/users/me/`. This will be appended to the `baseUrl` defined when [configuring the SDK object](./#configuring-a-sdk-object).
+*   `options`: The request initialization options that [`fetch()`](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch) accepts. Your [OAuth token](https://www.eventbrite.com/developer/v3/api_overview/authentication/#ebapi-getting-a-token) will be added to the request `Authorization` header for you. Some additional options you may need to pass in are:
+    *   `options.method`: The [HTTP method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) (e.g. `GET`, `POST`, etc.) for the fetch. Non-`GET` requests add `application/json` as `Content-Type` by default.
+    *   `options.mode`: The request mode (e.g. `cors`, `same-origin`, etc.) for the fetch. Defaults the `credentials` option to `include` when `mode` is `cors`. Otherwise the `credentials` default to `same-origin`.
+
+### Response
+
+The return value from `request()` is a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) that contains the response from the API call. If the response is successful, the response data will be parsed as JSON.
+
+See the [Error Handling](#error-handling) section for more information on the default error handling that `request()` provides.
+
+## Examples
```

Comment:
What do you think about giving a quick shoutout to the fact that you can use this sdk on the client or on the server? Not sure where it would live.. is it just assumed that libraries are using isomorphic fetch these days?

Reply:
I think I'll do this in #33 where the main README is getting updated. Think this is a good idea